### PR TITLE
bump egui to `0.30.0`

### PR DESCRIPTION
**Pull request by bircni**

### CODE REVIEW

**Comment by fluxxcode** [comment]
Thanks for the update. winit seems to be broken und linux though

**Comment by hacknus** [comment]
we need the `x11` feature on linux now, no?

**Comment by fluxxcode** [comment]
See my comment on: https://github.com/emilk/egui/issues/5493#issuecomment-2546708961

> According to emilk's comment: [#5489 (comment)](https://github.com/emilk/egui/pull/5489#issuecomment-2546362444) the `wayland` feature needs to be enabled for `eframe`. However, when the feature is enabled, I get the following compiling errors on Arch:
> 
> ```
> error: failed to select a version for the requirement `egui-wgpu = "^0.30.0"`
> candidate versions found which didn't match: 0.29.1, 0.29.0, 0.28.1, ...
> location searched: crates.io index
> required by package `eframe v0.30.0`
>     ... which satisfies dependency `eframe = "^0.30.0"` (locked to 0.30.0) of package `egui-file-dialog v0.7.0 (/home/jannis/projects/egui-file-dialog)`
> if you are looking for the prerelease package it needs to be specified explicitly
>     egui-wgpu = { version = "0.26.0-alpha.2" }
> ```
> 
> `egui-wgpu:v0.30.0` doesn't seem to be on crates.io yet

File: Cargo.toml [hunk 1]
```diff
@@ -22,19 +22,19 @@ sysinfo = { version = "0.33", default-features = false, features = ["disk"] }
...
 image-meta = { version = "0.1.2", optional = true }
-chrono = { version = "0.4.38", optional = true }
+chrono = { version = "0.4.39", optional = true }
 
 [dev-dependencies]
-eframe = { version = "0.29.1", default-features = false, features = [
+eframe = { version = "0.30.0", default-features = false, features = [
     "glow",
     "persistence",
 ] }
```
Comment (fluxxcode) [inline Cargo.toml:35]:
```suggestion
eframe = { version = "0.30.0", default-features = false, features = [
    "glow",
    "persistence",
    "wayland",
] }
```

**Review by fluxxcode — COMMENTED** [review]
I don't have permission to push on this branch, so here's a suggestion with the fix for Linux

**Comment by bircni** [comment]
The question is - do we wanna support Wayland or x11?

**Comment by hacknus** [comment]
shouldn't this be something that the user can select?

**Comment by fluxxcode** [comment]
We should support both

**Comment by fluxxcode** [comment]
updated my suggestion to also include x11

**Comment by fluxxcode** [comment]
> shouldn't this be something that the user can select?

Yes, this is just for the examples

**Comment by bircni** [comment]
@fluxxcode we could also set this as optional dependencies?

**Comment by fluxxcode** [comment]
No, I don't think that's helpful and rather confusing. `eframe` is only a dev dependency for the examples and is therefore not included in normal use of the library.
The examples should work on as many systems as possible, just like a normal egui app does.

**Comment by hacknus** [comment]
maybe document it in the examples `Cargo.toml` that this is required for linux?

**Comment by fluxxcode** [comment]
Will merge the PR now. Thanks!

**Pull request merged**